src/Console/Command/ProcessCommand.php

```diff
@@ -312,4 +307,53 @@ private function includeBootstrapFiles(): void
             }
         }
     }
+
+    /**
+     * @param SmartFileInfo[] $nonPhpFileInfos
+     */
+    private function runNonPhpFileProcessors(array $nonPhpFileInfos): void
+    {
+        foreach ($nonPhpFileInfos as $nonPhpFileInfo) {
+            foreach ($this->nonPhpFileProcessors as $nonPhpFileProcessor) {
+                if (! $nonPhpFileProcessor->canProcess($nonPhpFileInfo)) {
+                    continue;
+                }
+                $newContent = $nonPhpFileProcessor->process($nonPhpFileInfo);
+
+                if ($newContent === null) {
+                    continue;
+                }
+
+                $oldContent = $nonPhpFileProcessor->transformOldContent($nonPhpFileInfo);
+
+                $this->errorAndDiffCollector->addFileDiff($nonPhpFileInfo, $newContent, $oldContent);
+                if (! $this->configuration->isDryRun()) {
+                    $this->smartFileSystem->dumpFile($nonPhpFileInfo->getPathname(), $newContent);
+                    $this->smartFileSystem->chmod($nonPhpFileInfo->getRealPath(), $nonPhpFileInfo->getPerms());
+                }
+            }
+        }
+    }
+
+    /**
+     * @param string[] $paths
+     * @return SmartFileInfo[]
+     */
+    private function collectNonPhpFiles(array $paths): array
+    {
+        $nonPhpFileExtensions = [];
+        foreach ($this->nonPhpFileProcessors as $nonPhpFileProcessor) {
+            $nonPhpFileExtensions = array_merge($nonPhpFileProcessor->allowedFileExtensions(), $nonPhpFileExtensions);
+        }
+        $nonPhpFileExtensions = array_unique($nonPhpFileExtensions);
+
+        $nonPhpFileInfos = $this->filesFinder->findInDirectoriesAndFiles($paths, $nonPhpFileExtensions);
+
+        $composerJsonFilePath = getcwd() . '/composer.json';
+        if ($this->smartFileSystem->exists($composerJsonFilePath)) {
+            $nonPhpFileInfos[] = new SmartFileInfo($composerJsonFilePath);
+        }
+
+        return $nonPhpFileInfos;
+    }
```

Comment:
There seems lot of new logic in here. The `ProcessCommand` should be as thin as possible, as it's only command line wrapper. The application logic should be somewhere else, so it's useable also without command.

Saying that, could you decopule these 2 methods to own service?